larq_zoo/sota/quicknet.py

```diff
@@ -78,20 +86,70 @@ def residual_fast_block(
             activation="relu",
         )(x)
         x = tf.keras.layers.BatchNormalization(momentum=0.9, epsilon=1e-5)(x)
-        if downsample:
-            return tf.keras.layers.concatenate(
-                [residual, tf.keras.layers.add([x, residual])]
-            )
-        else:
-            return tf.keras.layers.add([x, residual])
+
+        if use_squeeze_and_excite:
+            x *= y
+
+        return x
+
+    def residual_block(self, x: tf.Tensor, use_squeeze_and_excite: bool) -> tf.Tensor:
+        """Standard residual block, without strides or filter changes."""
+        infilters = x.get_shape().as_list()[-1]
+        residual = x
+        x = self.conv_block(x, infilters, use_squeeze_and_excite)
+        return tf.keras.layers.add([x, residual])
+
+    def concat_transition_block(
+        self, x: tf.Tensor, filters: int, strides: int, use_squeeze_and_excite: bool
+    ) -> tf.Tensor:
+        """Concat transition block.
+
+        Doubles number of filters by concatenating shortcut with x + shortcut.
+        """
+        infilters = x.get_shape().as_list()[-1]
+        assert filters == 2 * infilters
+
+        residual = tf.keras.layers.MaxPool2D(pool_size=strides, strides=strides)(x)
+        residual = tf.keras.layers.BatchNormalization(momentum=0.9, epsilon=1e-5)(
+            residual
+        )
+        x = self.conv_block(x, infilters, use_squeeze_and_excite, strides)
+        x = tf.keras.layers.add([x, residual])
+
+        return tf.keras.layers.concatenate([residual, x])
+
+    def pointwise_transition_block(
```

Comment:
Can you please rename this block (or the description). The point of it is more akin to the improvement block in https://arxiv.org/abs/2001.05936

Reply:
The `pointwise` block is just an old-fashioned dimension-changing BiReal-Net block, I think you're talking about the `concat_transition_block`?

Reply:
I understand where the name comes from, however I think the renaming of it to something more instructive would be helpful - maybe ```fp_pointwise_transition_block``` or ```fp_improvement_transition_block```